src/connector/src/source/filesystem/opendal_source/opendal_reader.rs

```diff
@@ -76,24 +78,51 @@ impl<Src: OpendalSource> OpendalReader<Src> {
     #[try_stream(boxed, ok = StreamChunk, error = crate::error::ConnectorError)]
     async fn into_stream_inner(self) {
         for split in self.splits {
-            let data_stream = Self::stream_read_object(
-                self.connector.op.clone(),
-                split,
-                self.source_ctx.clone(),
-                self.connector.compression_format.clone(),
-            );
-
-            let data_stream = if need_nd_streaming(&self.parser_config.specific.encoding_config) {
-                nd_streaming::split_stream(data_stream)
+            let source_ctx = self.source_ctx.clone();
+
+            let object_name = split.name.clone();
+
+            let msg_stream;
+
+            if let EncodingProperties::Parquet = &self.parser_config.specific.encoding_config {
+                // // If the format is "parquet", use `ParquetParser` to convert `record_batch` into stream chunk.
+                let reader: tokio_util::compat::Compat<opendal::FuturesAsyncReader> = self
+                    .connector
+                    .op
+                    .reader_with(&object_name)
+                    .into_future() // Unlike `rustc`, `try_stream` seems require manual `into_future`.
+                    .await?
+                    .into_futures_async_read(split.offset as u64..)
```

Comment:
This is incorrect. `split.offset` refers to row offset not byte offset but opendal will take it as byte offset.